architecture-decision-records/0019-remove-frontend-messages-file.md

```diff
@@ -0,0 +1,63 @@
+# 19. Remove messages file from Transfer Digital Records frontend
+
+**Date:** 2021-06-07
+
+## Context
+
+The [TDR frontend project] currently contains a file called [messages.conf],
+which specifies all of the user-facing text shown by the frontend application.
+This includes text like titles, links, error messages and longer guidance. The
+HTML templates use the messages file by referencing a specific key. For example
+`@Messages("upload.continueLink")` is replaced by 'Start upload' when the
+template is rendered.
+
+The messages.conf file is a standard method of internationalising an app, and is
+[built into the Play framework][play-i18n]. We added it when we first started
+building TDR in late 2019 because we thought we might have to add multiple
+translations to the frontend.
+
+It now seems quite unlikely that we'll need to add translations soon, and we've
+encountered a few downsides to the messages file:
+
+* It makes it harder to update the HTML templates, because content is referred
+  to indirectly.
+* This is particularly difficult in content-heavy pages. See this [draft commit
```

Comment:
This second point is just an example of the difficulties caused by the first point. I would have this as a sub-point of the first point.

Reply:
Done.